homeassistant/components/ads/binary_sensor.py

```diff
@@ -34,14 +20,20 @@ def setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Binary Sensor platform for ADS."""
+    entities = []
+
+    if discovery_info is None:
+        return
+
     ads_hub = hass.data.get(DATA_ADS)
 
-    ads_var = config[CONF_ADS_VAR]
-    name = config[CONF_NAME]
-    device_class = config.get(CONF_DEVICE_CLASS)
+    for entry in discovery_info[CONF_BINARY_SENSORS]:
+        ads_var = entry.get(CONF_ADS_VAR)
```

Comment:
Please make sure to only use `dict.get` if the item is optional. Required dict items don't need that.

```suggestion
        ads_var = entry[CONF_ADS_VAR]
```